src/errors.rs

```diff
@@ -44,4 +44,6 @@ pub enum Error {
     ExpectedArrayValue(String),
     #[error("Expected an owned Object, but got `{0}`")]
     ExpectedObjectValue(Value),
+    #[error("Cannot convert GeoJson requested enum variant")]
+    GeoJsonWrongType,
```

Comment:
Can we use the existing `ExpectedType` instead? 

Or if there's a reason not to, can we add similar `expected` vs. `actual` so we can get a good error message like `ExpectedType` has?